ripple/BRRippleAccount.c

```diff
@@ -0,0 +1,317 @@
+
+#define WORD_LIST_LENGTH 2048
+
+struct BRRippleAccountRecord {
+    BRRippleAddress raw; // The 20 byte account id
+
+    // The public key - needed when sending 
+    BRKey publicKey;  // BIP44: 'Master Public Key 'M' (264 bits) - 8
+
+    uint32_t index;     // The BIP-44 Index used for this key.
+    
+    BRRippleSequence sequence;   // The NEXT valid sequence number, must be exactly 1 greater
+                                 // than the last transaction sent
+
+    BRRippleLastLedgerSequence lastLedgerSequence; // (Optional; strongly recommended) Highest ledger
+                                 // index this transaction
+                                 // can appear in. Specifying this field places a strict upper limit on
+                                 // how long the transaction can wait to be validated or rejected.
+                                 // See Reliable Transaction Submission for more details.
+};
+
+extern UInt512 getSeed(const char *paperKey)
+{
+    // Generate the 512bit private key using a BIP39 paperKey
+    UInt512 seed = UINT512_ZERO;
+    BRBIP39DeriveKey(seed.u8, paperKey, NULL); // no passphrase
+    return seed;
+}
+
+extern BRKey deriveRippleKeyFromSeed (UInt512 seed, uint32_t index)
+{
+    BRKey privateKey;
+    
+    // The BIP32 privateKey for m/44'/60'/0'/0/index
+    BRBIP32PrivKeyPath(&privateKey, &seed, sizeof(UInt512), 5,
+                       44 | BIP32_HARD,          // purpose  : BIP-44
+                       144 | BIP32_HARD,        // coin_type: Ripple
+                       0 | BIP32_HARD,          // account  : <n/a>
+                       0,                        // change   : not change
+                       index);                   // index    :
+    
+    privateKey.compressed = 0;
+    
+    return privateKey;
+}
+
+extern char * createRippleAddressString (BRRippleAddress address, int useChecksum)
+{
+    char *string = calloc (1, 36);
+
+    // The process is this:
+    // 1. Prepend the Ripple address indicator (0) to the 20 bytes
+    // 2. Do a douple sha265 hash on the bytes
+    // 3. Use the first 4 bytes of the hash as checksum and append to the bytes
+    uint8_t input[25];
+    input[0] = 0; // Ripple address type
+    memcpy(&input[1], address.bytes, 20);
+    uint8_t hash[32];
+    BRSHA256_2(hash, input, 21);
+    memcpy(&input[21], hash, 4);
+
+    // Now base58 encode the result
+    rippleEncodeBase58(string, 35, input, 25);
+    // NOTE: once the following function is "approved" we can switch to using it
+    // and remove the base58 function above
+    // static const char rippleAlphabet[] = "rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz";
+    // BRBase58EncodeEx(string, 36, input, 25, rippleAlphabet);
+    return string;
+}
+
+/**
+ * Optional way to create the BRKey
+ *
+ * 1. The normal way using the mnemonic paper_key
+ * 2. In DEBUG mode unit tests can pass in a private key instead
+ *
+ */
+BRKey getKey(const char* paperKey)
```

Comment:
Would splitting this into two functions (with one only exposed in debug builds) make sense from a testing point of view?